manual/estimator_manager.tex

```diff
@@ -0,0 +1,298 @@
+\section{Estimator Output}
+\subsection{Estimator Definition}
+For simplicity, consider a local property $O(\bs{R})$, where $\bs{R}$ is the collection of all particle coordinates. An \textit{estimator} for $O(\bs{R}) $ is a weighted average over walkers
+\begin{align}
+E[O] = \left(\sum\limits_{i=1}^{N^{tot}_{walker}} w_i O(\bs{R}_i) \right) / \left( \sum \limits_{i=1}^{N^{tot}_{walker}} w_i \right). \label{eq:estimator}
+\end{align}
+$N^{tot}_{walker}$ is the total number of walkers collected in the entire simulation. Notice, $N^{tot}_{walker}$ is typically far larger than the number of walkers held in memory at any given simulation step. $w_i$ is the weight of walker $i$.
+
+In a VMC simulation, the weight of every walkers is 1.0. Further, the number of walkers is constant at each step. Therefore, eq.~(\ref{eq:estimator}) simplifies to
+\begin{align}
+E_{VMC}[O] = \frac{1}{N_{step}N_{walker}^{ensemble}} \sum_{s,e} O(\bs{R}_{s,e}).
+\end{align}
+Each walker $\bs{R}_{s,e}$ is labeled by \textit{step index} s, and \textit{ensemble index} e.
+
+In a DMC simulation, the weight of each walker is different and may change from step to step. Further, the ensemble size varies from step to step. Therefore, eq.~(\ref{eq:estimator}) simplifies to
+\begin{align}
+E_{DMC}[O] = \frac{1}{N_{step}} \sum_{s} \left\{ \left(\sum_e w_{s,e} O(\bs{R}_{s,e})  \right) / \left( \sum \limits_{e} w_{s,e} \right)  \right\}.
+\end{align}
+
+I will refer to the average in the $\{\}$ as \textit{ensemble average} and the remaining averages \textit{block average}. The process of calculating $O(\bs{R})$ is \textit{evaluate}.
+
+\subsection{Class Relations}
+A large number of classes are involved in the estimator collection process. They often have misleading class name or method name. Document gotchas in the following list:
+\begin{enumerate}
+\item \verb|EstimatorManager| is an unused copy of \verb|EstimatorManagerBase|. \verb|EstimatorManagerBase| is the class used in the QMC drivers. (PR \#371 explains this)
+\item \verb|EstimatorManagerBase::Estimators| is completely different from \verb|QMCDriver::Estimators|, which is subtly different from \verb|QMCHamiltonianBase::Estimators|. The first is a list of pointers to \verb|ScalarEstimatorBase|. The second is the master estimator (one per MPI group). The third is the slave estimator that exists one per OpenMP thread.
+\item \verb|QMCHamiltonian| is NOT a parent class of \verb|QMCHamiltonianBase|. Instead, \verb|QMCHamiltonian| owns two lists of \verb|QMCHamiltonianBase| named \verb|H| and \verb|auxH|.

```

Comment:
Based on the names, I would (incorrectly) expect QMCHamiltonian to be derived from QMCHamiltonianBase.  It would be good to say what these are (or if they are defined in a previous section).  QMCHamiltonianBase is the base class for individual Hamiltonian pieces, and QMCHamiltonian is the container for all of them.  H is for Hamiltonian elements that contribute to the energy, and auxH is for elements that do not. (Someone else can comment if this description is wrong?)

Reply:
This is correct. A "HamiltonianBase" element is really a component of the Hamiltonian for ones that contribute to the energy and just an observable/operator otherwise.  The common denominator for HamiltonianBase is that each derived instance represents a quantum mechanical observable/operator.